.github/actions/deploy_branch/action.yml

```diff
@@ -75,15 +75,25 @@ runs:
         ingress_name="$RELEASE_NAME-data-claims-api"
         release_host="$RELEASE_NAME-$KUBE_NAMESPACE.cloud-platform.service.justice.gov.uk"
         identifier="$ingress_name-$KUBE_NAMESPACE-green"
+        branch_name=${{ steps.extract_release_name.outputs.branch-name }}
 
         echo "Deploying commit: $GIT_SHA under release name: $RELEASE_NAME to $release_host..."
+        
+        echo "branch name: $branch_name"
 
+        if [ "$branch_name" == "main" ]; then
+          export SPRING_PROFILE="main"
+        else
+          export SPRING_PROFILE="preview"
+        fi
```

Comment:
Remember, this file/action is only used when [deploying to uat](https://github.com/ministryofjustice/laa-data-claims-api/blob/main/.github/workflows/build-test-deploy.yml#L185), therefore would we need to replicate this on the `...deply/action.yml` as well?